### ISSUE TITLE
CentOS 7 hangs at "created API client, waiting for the control plane to become ready"

### ISSUE THREAD

**Issue by mikedanese** [kubernetes/kubeadm#39]
_From @samnissen on October 13, 2016 13:39_

**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.):
Possibly the same issue as #33729 -- different OS. 

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
Bug report

**Kubernetes version** (use `kubectl version`):

``` bash
Client Version: version.Info{Major:"1", Minor:"4", GitVersion:"v1.4.0", GitCommit:"a16c0a7f71a6f93c7e0f222d961f4675cd97a46b", GitTreeState:"clean", BuildDate:"2016-09-26T18:16:57Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"4", GitVersion:"v1.4.0", GitCommit:"a16c0a7f71a6f93c7e0f222d961f4675cd97a46b", GitTreeState:"clean", BuildDate:"2016-09-26T18:10:32Z", GoVersion:"go1.6.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Environment**:
- **Cloud provider or hardware configuration**: 
  privately hosted 4-core, 4gb VM
- **OS** (e.g. from /etc/os-release):

``` bash
NAME="CentOS Linux"
VERSION="7 (Core)"
```
- **Kernel** (e.g. `uname -a`):

``` bash
Linux abc-1234 3.10.0-327.10.1.el7.x86_64 #1 SMP Tue Feb 16 17:03:50 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

**What happened**:
I followed this guide precisely: http://kubernetes.io/docs/getting-started-guides/kubeadm/

At the step "(2/4) Initializing your master" the command `kubeadm init` stops at the line:

``` bash
<master/apiclient> created API client, waiting for the control plane to become ready
```

and never moves forward.

**What you expected to happen**:
The command would proceed as detailed in the guide.

**How to reproduce it** (as minimally and precisely as possible):
Follow the guide for CentOS 7 exactly.

**Anything else do we need to know**

_Copied from original issue: kubernetes/kubernetes#34720_

**Comment by mikedanese** [comment]
_From @pswenson on October 20, 2016 17:0_

I have the same issue.... trying to debug now

**Comment by mikedanese** [comment]
_From @jim-yefeng on October 25, 2016 8:14_

The same problem...

**Comment by mikedanese** [comment]
_From @errordeveloper on October 26, 2016 17:34_

Are all folks commenting here behind a proxy? If so, there is https://github.com/kubernetes/kubernetes/issues/34695 and thereby this is a duplicate.

**Comment by mikedanese** [comment]
_From @errordeveloper on October 26, 2016 17:35_

If you are using VritualBox and Vagrant, please take a look at the reference implementation here: https://github.com/errordeveloper/kubernetes-ansible-vagrant.

**Comment by mikedanese** [comment]
_From @madchap on October 26, 2016 19:58_

I am not behind a proxy, and have that issue (and using KVM)

**Comment by mikedanese** [comment]
_From @samnissen on October 27, 2016 9:52_

Also not behind proxy, and on KVM

**Comment by mikedanese** [comment]
_From @smalllove1459 on October 27, 2016 10:56_

Using a Proxy and stuck here for hours...

**Comment by k8sdude88** [comment]
I had this issue on a DigitalOcean droplet running Debian Jessie.  No proxy issues etc.

It turns out that in my case the kubelet refused to start containers due to memory accounting cgroup not being enabled.

To fix it I did the old standby: edited `/etc/default/grub` line `GRUB_CMDLINE_LINUX` to include `cgroup_enable=memory` arg and rebooted the VM.  Then running `kubeadm init` from scratch worked.

@lukemarsden and others, you might want to consider adding a pre-flight check for this condition (`docker info` seems to divine it somehow).  In my case, it just hung and I had to go digging through many logs and `kubeadm` architecture to figure out why.

**Comment by luxas** [comment]
We do know it, and this is exactly why we've chosen to not support debian yet, since it isn't presented in a good way. We chose early on to use sig-node's system verification check, but it wasn't ready before v1.5.

Now we have that working as well, and my PR for that will soon be merged: https://github.com/kubernetes/kubernetes/pull/36334
The next revision released of kubeadm will include this patch.

**Comment by luxas** [comment]
Please try v1.6 and open a new issue if this error can be reproduced...

**Comment by UmarSaeed11** [comment]
I am having the same problem. I have installed 3 Centos 7 machines on virtualbox and had made one of them as master by following " https://kubernetes.io/docs/setup/independent/create-cluster-kubeadm/ " guide. Now when I hit " kubeadm init ", the process stucks at "  Created API client, waiting for the control plane to become ready ". 

Kindly help me out as I am new to kubernetes and have to deploy a cluster for my office. Thank you!

**Comment by kachkaev** [comment]
@UmarSaeed11 this problem has been mentioned in a few discussion threads, the info is quite chaotic. I also struggled with it about a month ago and was able to get through by configuring dnsmasq. See https://github.com/kubernetes/kubeadm/issues/228#issuecomment-307158412 and the following comments – hope it helps.

This might be the cause: https://github.com/kubernetes/kubeadm/issues/302

**Comment by UmarSaeed11** [comment]
Thanks for your response Alexander!

I followed the steps but still it is stuck on " [apiclient] Created API
client, waiting for the control plane to become ready ".

I am using virtualbox and created 3 Centos 7 virtual machines. I am able to
ping each machine from every other virtual machine. Is this the right way
to test kubernetes?

Moreover, I have also disabled iptables and firewalld.

Regards,
@UmarSaeed11 <https://github.com/umarsaeed11>
